jasc/BUILD

```diff
@@ -0,0 +1,223 @@
+# Schedules for JAX.
+
+load("@rules_python//python:defs.bzl", "py_library")
+load("@pybind11_bazel//:build_defs.bzl", "pybind_extension")
+load("@llvm-project//mlir:build_defs.bzl", "cc_headers_only")
+
+package(
+    # default_applicable_licenses = ["//third_party/mlir_edge:license"],
+    default_visibility = ["//visibility:public"],
+)
+
+py_library(
+    name = "jasc",
+    srcs = ["jasc.py", "__init__.py"],
+    deps = [
+        ":call_kernel",
+        ":primitives",
+        "//dialect:python",
+        "//transform_ops",
+        "@jax1//jax:jax",
+        "@jax1//jaxlib/mlir:bufferization_dialect",
+        "@jax1//jaxlib/mlir:core",
+        "@jax1//jaxlib/mlir:ir",
+        "@jax1//jaxlib/mlir:pdl_dialect",
+        "@jax1//jaxlib/mlir:transform_dialect",
+        "@jax1//jaxlib/mlir:jasc_dialect",
+    ],
+)
+
+py_library(
+    name = "tuner",
+    srcs = ["tuner.py"],
+    deps = [
+        ":jasc",
+        "@jax1//jax:jax",
+        "@jax1//jaxlib/mlir:ir",
+        "@jax1//jaxlib/mlir:jasc_dialect",
+        "@jax1//jaxlib/mlir:transform_dialect",
+    ],
+)
+
+py_library(
+    name = "primitives",
+    srcs = ["primitives.py"],
+    deps = [
+        ":call_kernel",
+        "//dialect:python",
+        "@jax1//jax:jax",
+        "@jax1//jax:extend",
+        "@jax1//jaxlib/mlir:ir",
+        "@jax1//jaxlib/mlir:pdl_dialect",
+        "@jax1//jaxlib/mlir:stablehlo_dialect",
+        "@jax1//jaxlib/mlir:transform_dialect",
+    ],
+)
+
+cc_library(
+    name = "call_kernel_shared_library_deps",
+    deps = [
+        "@jax1//jaxlib/mlir/_mlir_libs:jaxlib_mlir_capi_shared_library",
+        ":mlir_lowering_shared_library",
+        # "//third_party/gpus/cuda:cuda_headers",
+        "@llvm-project//llvm:Support",
+        "@llvm-project//mlir:CAPIIR",
+        "@llvm-project//mlir:ExecutionEngine",
+        "@llvm-project//mlir:FuncDialect",
+        "@llvm-project//mlir:IR",
+        "@llvm-project//mlir:MLIRBindingsPythonHeaders",
+        "@pybind11_abseil//pybind11_abseil:import_status_module",
+        "@pybind11_abseil//pybind11_abseil:status_casters",
+    ],
+)
+
+cc_headers_only(
+    name = "call_kernel_shared_library_deps_headers",
+    src = "call_kernel_shared_library_deps",
+)
+
+cc_binary(
+    name = "libcallkernel.so",
+    linkopts = [
+        "-Wl,-soname=libcallkernel.so",
+        "-Wl,-rpath='$$ORIGIN'",
+    ],
```

Comment:
Nit: any comment on what this does and why is helpful to debug this in the future.

Reply:
I don't know *exactly* what this does. I have copied that pattern from JAX's BUILD files. The latest commit contains a comment with what I know and where the code comes from.